browser/net/brave_site_hacks_network_delegate_helper_unittest.cc

```diff
@@ -267,4 +268,86 @@ TEST_F(BraveSiteHacksNetworkDelegateHelperTest,
   });
 }
 
+TEST_F(BraveSiteHacksNetworkDelegateHelperTest, QueryStringUntouched) {
+  std::vector<GURL> urls({
+    GURL("https://example.com/"),
+    GURL("https://example.com/?"),
+    GURL("https://example.com/?+%20"),
+    GURL("https://user:pass@example.com/path/file.html?foo=1#fragment"),
+    GURL("http://user:pass@example.com/path/file.html?foo=1&bar=2#fragment"),
+    GURL("https://example.com/?file=https%3A%2F%2Fexample.com%2Ftest.pdf"),
+    GURL("https://example.com/?title=1+2&caption=1%202"),
+    GURL("https://example.com/?foo=1&&bar=2#fragment"),
+    GURL("https://example.com/?foo&bar=&#fragment"),
+    GURL("https://example.com/?foo=1&fbcid=no&gcid=no&mc_cid=no&bar=&#frag"),
+    GURL("https://example.com/?fbclid=&gclid&=mc_eid&msclkid="),
+    GURL("https://example.com/?value=fbclid=1&not-gclid=2&foo+mc_eid=3"),
+    GURL("https://example.com/?+fbclid=1"),
+    GURL("https://example.com/?%20fbclid=1"),
+    GURL("https://example.com/#fbclid=1"),
+  });
+  std::for_each(urls.begin(), urls.end(), [this](GURL url){
+    net::TestDelegate test_delegate;
+    std::unique_ptr<net::URLRequest> request =
+        context()->CreateRequest(url, net::IDLE, &test_delegate,
+                                 TRAFFIC_ANNOTATION_FOR_TESTS);
+
+    std::shared_ptr<brave::BraveRequestInfo>
+        brave_request_info(new brave::BraveRequestInfo());
+    brave::BraveRequestInfo::FillCTXFromRequest(request.get(),
+        brave_request_info);
+    brave::ResponseCallback callback;
+    int ret = brave::OnBeforeURLRequest_SiteHacksWork(callback,
+        brave_request_info);
+    EXPECT_EQ(ret, net::OK);
+    // new_url should not be set
+    EXPECT_TRUE(brave_request_info->new_url_spec.empty());
+    EXPECT_EQ(request->url(), url);
+  });
+}
+
+TEST_F(BraveSiteHacksNetworkDelegateHelperTest, QueryStringFiltered) {
+  std::vector<std::pair<GURL, GURL> > urls({
```

Comment:
I'd write `const auto urls = {...};`

Reply:
I wasn't able to make that work with auto, but I did change it to remove the `GURL`s.